File: kotlin/kindergarten-garden/src/main/kotlin/KindergartenGarden.kt
class KindergartenGarden(private val diagram: String) {
    companion object {
        private val students = listOf( "Alice", "Bob", "Charlie", "David", "Eve", "Fred", "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry" )
        private val plants = mapOf( 'G' to "grass", 'C' to "clover", 'R' to "radishes", 'V' to "violets" )
    }

    fun getPlantsOfStudent(student: String) = diagram.split("\n")
        .map { it.chunked(2)[students.indexOf(student)] }.flatMap { it.toList() }.map { plants[it]!! }
}

File: csharp/kindergarten-garden/KindergartenGarden.cs
using System;
using System.Collections.Generic;
using System.Linq;

public enum Plant
{
    Violets = 'V',
    Radishes = 'R',
    Clover = 'C',
    Grass = 'G'
}

public class KindergartenGarden
{
    private readonly string diagram;

    private readonly List<string> studentsName = new List<string>()
    {
        "Alice", "Bob", "Charlie", "David",
        "Eve", "Fred", "Ginny", "Harriet",
        "Ileana", "Joseph", "Kincaid", "Larry"
    };

    public KindergartenGarden(string diagram)
    {
        this.diagram = diagram;
    }

    public IEnumerable<Plant> Plants(string student)
    {
        int index = studentsName.IndexOf(student) * 2;
        return from line in diagram.Split('\n')
            from idx in Enumerable.Range(index, 2)
            select (Plant)line[idx];
    }
}
